// office-ui-fabric-react/lib/components/HoverCard/HoverCard.doc.js
import * as React from 'react';
import { HoverCardBasicExample } from './examples/HoverCard.Basic.Example';
import { HoverCardTargetExample } from './examples/HoverCard.Target.Example';
import { HoverCardStatus } from './HoverCard.checklist';
import './HoverCardPage.global.scss';
var HoverCardBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/examples/HoverCard.Basic.Example.tsx');
var HoverCardTargetExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/examples/HoverCard.Target.Example.tsx');
export var HoverCardPageProps = {
    title: 'HoverCard',
    componentName: 'HoverCard',
    componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/HoverCard',
    componentStatus: HoverCardStatus,
    examples: [
        {
            title: 'Example 1: HoverCard wrapping an element',
            code: HoverCardBasicExampleCode,
            view: React.createElement(HoverCardBasicExample, null)
        },
        {
            title: 'Example 2: HoverCard using Target, DirectionalHint and custom HotKey',
            code: HoverCardTargetExampleCode,
            view: React.createElement(HoverCardTargetExample, null)
        }
    ],
    propertiesTablesSources: [
        require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/HoverCard.types.ts'),
        require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/ExpandingCard.types.ts')
    ],
    overview: require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/docs/HoverCardOverview.md'),
    bestPractices: 'Use this guidance when need to fine tune properties of HoverCard',
    dos: require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/docs/HoverCardDos.md'),
    donts: require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/docs/HoverCardDonts.md'),
    isHeaderVisible: true,
    isFeedbackVisible: true,
    allowNativeProps: true
};
