// office-ui-fabric-react/lib-amd/components/HoverCard/HoverCard.doc.js
define(["require", "exports", "react", "./examples/HoverCard.Basic.Example", "./examples/HoverCard.Target.Example", "./HoverCard.checklist", "./HoverCardPage.global.scss"], function (require, exports, React, HoverCard_Basic_Example_1, HoverCard_Target_Example_1, HoverCard_checklist_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var HoverCardBasicExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/examples/HoverCard.Basic.Example.tsx');
    var HoverCardTargetExampleCode = require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/examples/HoverCard.Target.Example.tsx');
    exports.HoverCardPageProps = {
        title: 'HoverCard',
        componentName: 'HoverCard',
        componentUrl: 'https://github.com/OfficeDev/office-ui-fabric-react/tree/master/packages/office-ui-fabric-react/src/components/HoverCard',
        componentStatus: HoverCard_checklist_1.HoverCardStatus,
        examples: [
            {
                title: 'Example 1: HoverCard wrapping an element',
                code: HoverCardBasicExampleCode,
                view: React.createElement(HoverCard_Basic_Example_1.HoverCardBasicExample, null)
            },
            {
                title: 'Example 2: HoverCard using Target, DirectionalHint and custom HotKey',
                code: HoverCardTargetExampleCode,
                view: React.createElement(HoverCard_Target_Example_1.HoverCardTargetExample, null)
            }
        ],
        propertiesTablesSources: [
            require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/HoverCard.types.ts'),
            require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/ExpandingCard.types.ts')
        ],
        overview: require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/docs/HoverCardOverview.md'),
        bestPractices: 'Use this guidance when need to fine tune properties of HoverCard',
        dos: require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/docs/HoverCardDos.md'),
        donts: require('!raw-loader!office-ui-fabric-react/src/components/HoverCard/docs/HoverCardDonts.md'),
        isHeaderVisible: true,
        isFeedbackVisible: true,
        allowNativeProps: true
    };
});
